### ISSUE TITLE
Feature request: support differing casing conventions

### ISSUE THREAD

**Issue by michaelbridge** [typed-graphql-builder/typed-graphql-builder#62]
Thank you. This is a really phenomenal library -- I can't believe it's not seen more traffic! There is no comparable library that properly handles query and response typing consistently.

There's a common use case where the GraphQL schema has a different casing convention than the front-end (i.e., `snake_case` vs `camelCase`). GraphQL Codegen handles this via the [naming convention](https://the-guild.dev/graphql/codegen/docs/config-reference/naming-convention) configuration option. Do you think this is something that would ever be supported? Possibly related to #27. 

I started to look at writing a library wrapper to handle this, but the typing gets tricky quickly (due to the extent of the generated code). Any thoughts/suggestions here?

Thank you again.

**Comment by spion** [comment]
My naive first thought is that we should be able to include the correct case-conversion function in the output, then use it in the graphql query builder function only to process all inputs and outputs. However, there are few things that could be problematic: 

- nested JSON scalars for input types will need to be skipped unless there is some encoding of which input fields are json scalars. This can probably be done by utilizing argTypes, although that whole function needs to be cleaned up before adding complexity to it

- typed graphql builder is just a query builder. It cannot by design process the output returned from the server, so it cannot make any output type corrections or conversions to it.

I think the second part is what really makes this idea impossible within the context of this library. I'm curious however if some kind of client and schema conversion middleware would be able to do this in a library-independent way

**Comment by michaelbridge** [comment]
Agreed this would be a big lift. I experimented with wrapping the generated client. #2 is quite doable but #1 proved tricky to implement in all circumstances (especially for nested input JSON args). I ultimately abandoned the effort since it was difficult for a single use case, let alone as a generalized wrapper library.
